assertsprocessor/utils.go

```diff
@@ -0,0 +1,123 @@
+package assertsprocessor
+
+import (
+	"context"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap"
+	"regexp"
+)
+
+func buildEntityKey(config *Config, namespace string, service string) EntityKeyDto {
+	return EntityKeyDto{
+		Type: "Service",
+		Name: service,
+		Scope: map[string]string{
+			"env": config.Env, "site": config.Site, "namespace": namespace,
+		},
+	}
+}
+
+func computeLatency(span ptrace.Span) float64 {
+	return float64(span.EndTimestamp()-span.StartTimestamp()) / 1e9
+}
+
+func compileRequestContextRegexps(logger *zap.Logger, config *Config) (*map[string]regexp.Regexp, error) {
+	logger.Info("consumer.Start compiling request context regexps")
+	var exps = map[string]regexp.Regexp{}
+	if config.RequestContextExps != nil {
+		for attName, matchExpString := range *config.RequestContextExps {
+			compile, err := regexp.Compile(matchExpString)
+			if err != nil {
+				return nil, err
+			}
+			exps[attName] = *compile
```

Comment:
This is value copying. `Regexp` seems expensive, so perhaps the value copying here can be replaced with a pointer copying? Though in that case, the returned map needs to be a string-to-pointer map.

I have the same concern about the `attributeValueRegExps` field in `metricHelper`.

Reply:
> This is value copying. `Regexp` seems expensive, so perhaps the value copying here can be replaced with a pointer copying? Though in that case, the returned map needs to be a string-to-pointer map.
> 
> I have the same concern about the `attributeValueRegExps` field in `metricHelper`.

Addressed